### ISSUE TITLE
Bundler 1.10.3 resolves dependencies indefinitely.

### ISSUE THREAD

**Issue by adammathys** [bundler/bundler#3732]
I'm currently unable to use Bundler 1.10 on most projects due to it taking forever to resolve dependencies.

Output of `bundle env` (with some minor data scrubbing):
```shell
Environment

    Bundler   1.10.3
    Rubygems  2.4.8
    Ruby      2.1.6p336 (2015-04-13 revision 50298) [x86_64-linux]
    GEM_HOME  /home/user/.gem/ruby/2.1.6
    GEM_PATH  /home/user/.gem/ruby/2.1.6:/home/user/.rubies/ruby-2.1.6/lib/ruby/gems/2.1.0
    Git       1.9.1

Bundler settings

    path
      Set for the current user (/home/user/.bundle/config): "vendor/bundle"
    disable_shared_gems
      Set for your local app (/home/user/Work/project/.bundle/config): "1"
      Set for the current user (/home/user/.bundle/config): "1"

Gemfile

    source 'https://rubygems.org'
    
    # Provides basic authentication functionality for testing parts of your engine
    gem 'spree', github: 'spree/spree', branch: '2-2-stable'
    gem 'spree_auth_devise', github: 'spree/spree_auth_devise', branch: '2-2-stable'
    gem 'byebug'
    
    gemspec

Gemfile.lock

    <No /home/user/Work/project/Gemfile.lock found>
```

Result of running `bundle install` for ~11 minutes:
```shell
Fetching git://github.com/spree/spree.git
Fetching git://github.com/spree/spree_auth_devise.git
Fetching gem metadata from https://rubygems.org/.............
Fetching version metadata from https://rubygems.org/...
Fetching dependency metadata from https://rubygems.org/..
Resolving dependencies.......................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................^C
bundle install  674.65s user 3.11s system 98% cpu 11:29.16 total
```

Reverting to 1.9.9 fixes the issue. I tested and was able to reproduce the error with versions 1.10.1, 1.10.2, and 1.10.3.

**Comment by segiddins** [comment]
What's the content of the gem spec, so I can attempt to debug this?

**Comment by adammathys** [comment]
Here's the trimmed down `.gemspec`:
```ruby
# encoding: UTF-8
Gem::Specification.new do |s|
  s.platform    = Gem::Platform::RUBY
  s.name        = 'gem_name'
  s.version     = '2.2.1'
  s.required_ruby_version = '>= 2.1'

  s.require_path = 'lib'
  s.requirements << 'none'

  s.add_dependency 'spree', '~> 2.2.1'
  s.add_development_dependency 'rspec-rails', '~> 2.13'
end
```

I went through and removed as much as possible while it still failed. I believe this is the minimum `.gemspec` to reproduce the error.

**Comment by segiddins** [comment]
Thanks! I have a hunch why this is happening, but I'm slammed for the next few days, I'll try and get around to this when I can. 

-Samuel E. Giddins

On Jun 9, 2015, at 2:36 PM, Adam Mueller <notifications@github.com> wrote:

Here's the trimmed down .gemspec:

# encoding: UTF-8
Gem::Specification.new do |s|
  s.platform    = Gem::Platform::RUBY
  s.name        = 'gem_name'
  s.version     = '2.2.1'
  s.required_ruby_version = '>= 2.1'

  s.require_path = 'lib'
  s.requirements << 'none'

  s.add_dependency 'spree', '~> 2.2.1'
  s.add_development_dependency 'rspec-rails', '~> 2.13'
end
I went through and removed as much as possible while it still failed. I believe this is the minimum .gemspec to reproduce the error.

**Comment by adammathys** [comment]
Awesome! Thanks. :smile:

**Comment by andremedeiros** [comment]
Closing this off seen as it's been fixed by #3736 already.
